configs/components/curl.rb

```diff
@@ -4,18 +4,44 @@
   pkg.url "http://buildsources.delivery.puppetlabs.net/curl-#{pkg.get_version}.tar.gz"
 
   pkg.build_requires "openssl"
+  pkg.build_requires "puppet-ca-bundle"
+
+  prefix = settings[:prefix]
+  pem_file = "#{settings[:prefix]}/ssl/cert.pem"
+
+  make = platform[:make]
+
+  if platform.is_windows?
+    pkg.build_requires "pl-zlib-#{platform.architecture}"
+    pkg.build_requires "runtime"
+
+    pkg.environment "PATH" => "#{settings[:gcc_bindir]}:$$PATH"
+    pkg.environment "CYGWIN" => settings[:cygwin]
+    pkg.environment "CC" => settings[:cc]
+    pkg.environment "CXX" => settings[:cxx]
+
+    prefix = platform.convert_to_windows_path(settings[:prefix])
+    pem_file = platform.convert_to_windows_path("#{settings[:puppet_configdir]}/ssl/cert.pem")
```

Comment:
I would probably give it a more descriptive name.

However, I don't see anything it would conflict with. Nothing at https://github.com/puppetlabs/puppet/blob/master/lib/puppet/defaults.rb#L645-L770 does.